connector/github/github.go

```diff
@@ -233,37 +278,101 @@ func (c *githubConnector) HandleCallback(s connector.Scopes, r *http.Request) (i
 	return identity, nil
 }
 
-func (c *githubConnector) Refresh(ctx context.Context, s connector.Scopes, ident connector.Identity) (connector.Identity, error) {
-	if len(ident.ConnectorData) == 0 {
-		return ident, errors.New("no upstream access token found")
+func (c *githubConnector) Refresh(ctx context.Context, s connector.Scopes, identity connector.Identity) (connector.Identity, error) {
+	if len(identity.ConnectorData) == 0 {
+		return identity, errors.New("no upstream access token found")
 	}
 
 	var data connectorData
-	if err := json.Unmarshal(ident.ConnectorData, &data); err != nil {
-		return ident, fmt.Errorf("github: unmarshal access token: %v", err)
+	if err := json.Unmarshal(identity.ConnectorData, &data); err != nil {
+		return identity, fmt.Errorf("github: unmarshal access token: %v", err)
 	}
 
 	client := c.oauth2Config(s).Client(ctx, &oauth2.Token{AccessToken: data.AccessToken})
 	user, err := c.user(ctx, client)
 	if err != nil {
-		return ident, fmt.Errorf("github: get user: %v", err)
+		return identity, fmt.Errorf("github: get user: %v", err)
 	}
 
 	username := user.Name
 	if username == "" {
 		username = user.Login
 	}
-	ident.Username = username
-	ident.Email = user.Email
+	identity.Username = username
+	identity.Email = user.Email
 
-	if s.Groups && c.org != "" {
-		groups, err := c.teams(ctx, client, c.org)
+	if s.Groups && len(c.orgs) > 0 {
+		groups, inOrgNoTeams, err := c.listGroups(ctx, client, username)
 		if err != nil {
-			return ident, fmt.Errorf("github: get teams: %v", err)
+			return identity, err
+		}
+		if len(groups) == 0 && !inOrgNoTeams {
+			return identity, fmt.Errorf("github: user %q not a member of any orgs or teams provided", username)
 		}
-		ident.Groups = groups
+		identity.Groups = groups
 	}
-	return ident, nil
+
+	return identity, nil
+}
+
+// listGroups enforces org and team constraints on user authorization
+// Cases in which user is authorized:
+// 	N orgs, no teams: user is member of at least 1 org
+// 	N orgs, M teams per org: user is member of any team from at least 1 org
+// 	N-1 orgs, M teams per org, 1 org with no teams: user is member of any team
+// from at least 1 org, or member of org with no teams
+func (c *githubConnector) listGroups(ctx context.Context, client *http.Client, userName string) ([]string, bool, error) {
+	var (
+		inOrg, inOrgNoTeams bool
+		teams, groups       []string
```

Comment:
`inOrg`, `teams`, and `err` do not need to be defined outside the for loop. Just define them as used in the for loop,